symengine/expand.cpp

```diff
@@ -0,0 +1,317 @@
+#include <symengine/basic.h>
+#include <symengine/visitor.h>
+#include <symengine/pow.h>
+
+namespace SymEngine {
+
+inline RCP<const Number> _mulnum(const RCP<const Number> &x, const RCP<const Number> &y) {
+    if (eq(*x, *one)) return y;
+    if (eq(*y, *one)) return x;
+    return x->mul(*y);
+}
+
+inline void _imulnum(const Ptr<RCP<const Number>> &self, const RCP<const Number> &other) {
+    *self = _mulnum(*self, other);
+}
+
+class ExpandVisitor : public BaseVisitor<ExpandVisitor> {
+public:
+    umap_basic_num d_;
+    RCP<const Number> coeff = zero;
+    RCP<const Number> multiply = one;
+
+    ExpandVisitor() : BaseVisitor<ExpandVisitor>(this) {
+
+    }
+
+    void apply(const RCP<const Basic> &b) {
+        b->accept(*this);
+    }
```

Comment:
It was used at https://github.com/symengine/symengine/pull/668/files#diff-3ca70833b804e0d56ab69290843badb8R48,.
There were more uses in previous work, but now that there's only one place I can use the method without RCP and delete this method.